SwanProjects/swanprojects/handlers.py

```diff
@@ -0,0 +1,289 @@
+            self.finish(json.dumps(data))
+            return
+
+        try:
+            swan_user_script_file = project_dir + os.path.sep + '.userscript'
+            with open(swan_user_script_file, 'w') as f:
+                f.write(user_script)
+                f.close()
+        except Exception as msg:
+            data = {"status": False, "project_dir": f"SWAN_projects/{name}",
+                    "msg": f"Error creating .userscript file for project {name}, traceback: {msg}"}
+            self.finish(json.dumps(data))
+            return
+
+        command = get_env_isolated()
+        command += ["/bin/bash", "-c", "swan_kmspecs --project_name %s" % name]
+        self.log.info(f"running {command} ")
+        proc = subprocess.Popen(command, stdout=subprocess.PIPE)
+        proc.wait()
+        output = proc.stdout.read().decode("utf-8")
+        self.log.info(f"swan_kmspecs output: {output}")
+        proc.communicate()
+        self.log.info(f"swan_kmspecs return code: {proc.returncode}")
+        if proc.returncode != 0:
+            data = {"status": False, "project_dir": f"SWAN_projects/{name}",
+                    "msg": f"Error collecting the information from cvmfs for project {name},  traceback: {output}"}
+            self.finish(json.dumps(data))
+            return
+
+        data = {"status": True, "project_dir": f"SWAN_projects/{name}",
+                "msg": f"created project {name}"}
+        self.finish(json.dumps(data))
+
+class EditProjectHandler(APIHandler):
+
+    @tornado.web.authenticated
+    def post(self):
+        """
+        This endpoint allows to edit project information, such as name, stack, platform etc..
+        The project can be renamed from $HOME/SWAN_projects/old_name to $HOME/SWAN_projects/name
+        and metadata in the .swanproject is updated.
+
+        This endpoint allows also edit corrupted projects, if something is wrong in .swanproject
+        the edit project dialog will send the information again to this endpoint to fix the project information.
+        """
+        input_data = self.get_json_body()
+        print(f"EditProjectHandler = {input_data}")
+
+        corrupted = False
+        if "corrupted" in input_data.keys():
+            corrupted = input_data["corrupted"]
+
+        old_name = ""
+        if "old_name" in input_data.keys():
+            old_name = input_data["old_name"]
+
+        old_stack = ""
+        if "old_stack" in input_data.keys():
+            old_stack = input_data["old_stack"]
+
+        old_platform = ""
+        if "old_platform" in input_data.keys():
+            old_platform = input_data["old_platform"]
+
+        old_release = ""
+        if "old_release" in input_data.keys():
+            old_release = input_data["old_release"]
+
+        old_userscript = ""
+        if "old_userscript" in input_data.keys():
+            old_userscript = input_data["old_userscript"]
```

Comment:
Each of these could be simplified to:
```python
old_attr = input_data.get("old_attr")
```
`old_attr` would be `None` if the key is not in the dictionary.

Reply:
And instead of hardcoding the values, can't you just traverse the original dict and check if there's a replacement? A merge dict or smth like that.

Reply:
Simplified.